src/page/util.rs

```diff
@@ -244,33 +269,123 @@ where
         }
     }
 
-    let mut result: Vec<IndexData<T>> = vec![];
+    let mut result: Vec<IndexValue<T>> = vec![];
+    for interval in intervals.iter() {
+        for index in interval.0..=interval.1 {
+            let mut index_records = parse_index_page::<T, PAGE_SIZE>(file, index as u32)?;
```

Comment:
Hm, it looks like for every `Link` we are currently seeking at page start and re-reading header and then seeking to data start. I think it's not the best way to do that